week07/manager_test.go

```diff
@@ -0,0 +1,210 @@
+		ctx   context.Context
+		count int
+	}
+
+	tests := []struct {
+		name   string
+		args   args
+		expErr bool
+	}{
+		{name: "with valid employee count", args: args{ctx: context.Background(), count: 2}, expErr: false},
+		{name: "with invalid employee count", args: args{ctx: context.Background(), count: 0}, expErr: true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := m.Start(tt.args.ctx, tt.args.count)
+
+			actErr := (err != nil)
+
+			if actErr != tt.expErr {
+				t.Fatalf("got err %v, expected err %v", err, tt.expErr)
+			}
+		})
+	}
+
+}
+
+func Test_Assign(t *testing.T) {
+	t.Parallel()
+
+	t.Run("with invalid products", func(t *testing.T) {
+		p := []*Product{
+			{Quantity: 0},
+		}
+
+		m := NewManager()
+		err := m.Assign(p...)
+		if err == nil {
+			t.Fatal("expected err got none")
+		}
+	})
+
+	t.Run("with stopped manager", func(t *testing.T) {
+		p := []*Product{
+			{Quantity: 1},
+			{Quantity: 3},
+		}
+
+		m := NewManager()
+		m.Stop()
+		err := m.Assign(p...)
+		if err == nil {
+			t.Fatal("expected err got none")
+		}
+
+	})
+}
+
+func Test_Complete(t *testing.T) {
+	m := NewManager()
+	type args struct {
+		e Employee
+		p *Product
+	}
+
+	tests := []struct {
+		name   string
+		args   args
+		expErr bool
+	}{
+		{name: "with invalid employee", args: args{e: Employee(0), p: &Product{Quantity: 1}}, expErr: true},
+		{name: "with invalid product", args: args{e: Employee(1), p: &Product{Quantity: 0}}, expErr: true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := m.Complete(tt.args.e, tt.args.p)
+
+			act := (err != nil)
+
+			if act != tt.expErr {
+				t.Fatalf("got err %v, expected err %v", err, tt.expErr)
+			}
+
+		})
+	}
+}
+
+// TODO: Implement test cases for the Run function.
+func Test_Run(t *testing.T) {
+	t.Parallel()
+	p := []*Product{
+		{Quantity: 100},
+		{Quantity: 300},
+	}
+	e := 5
+
+	// Tests you will need to write:
+
+	// TODO: timeout after 5 seconds if nothing happens
+	t.Run("with 5 second timeout", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+		defer cancel()
+
+		_, err := Run(ctx, e, p...)
+		if err != nil {
+			t.Fatalf("unexpected err: %v", err)
+		}
+
+		time.Sleep(time.Second * 6)
+
+		<-ctx.Done()
+
+		if ctx.Err() != context.DeadlineExceeded {
+			t.Fatalf("got context error: %v, expected %v", ctx.Err(), context.DeadlineExceeded)
+		}
+	})
+
+	t.Run("with invalid employees", func(t *testing.T) {
+		_, err := Run(context.Background(), 0, p...)
+		if err == nil {
+			t.Fatal("expected error received none")
+		}
+	})
+
+	t.Run("with invalid products", func(t *testing.T) {
+		ip := []*Product{
+			{Quantity: 0},
+		}
+		_, err := Run(context.Background(), e, ip...)
+		if err == nil {
+			t.Fatal("expected error received none")
+		}
+
+	})
+
+	// TODO: interruption by a signal
+	t.Run("with interrupt signal", func(t *testing.T) {
+		const TEST_SIGNAL = syscall.SIGINT
```

Comment:
Try using `SIGUSR1` to fix the windows problem. If that doesn't work. mark the test with `t.Skip()` so it doesn't keep failing in the future. It's OK to skip this test and move on.

```suggestion
		const TEST_SIGNAL = syscall.SIGUSR1
```

Reply:
`SIGUSR1` came back undefined, so skipping and commenting out the test. Thank you for the feedback